akka-typed/src/main/scala/akka/typed/internal/ActorSystemImpl.scala

```diff
@@ -0,0 +1,251 @@
+                err.print(thread.getName)
+                err.print("] shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[")
+                err.print(name)
+                err.println("]")
+                cause.printStackTrace(System.err)
+                System.err.flush()
+              } finally {
+                System.exit(-1)
+              }
+            } else {
+              log.error(cause, "Uncaught fatal error from thread [{}] shutting down ActorSystem [{}]", thread.getName, name)
+              terminate()
+            }
+        }
+      }
+    }
+
+  override val threadFactory: d.MonitorableThreadFactory =
+    d.MonitorableThreadFactory(name, settings.Daemonicity, Option(_cl), uncaughtExceptionHandler)
+
+  override val dynamicAccess: a.DynamicAccess = new a.ReflectiveDynamicAccess(_cl)
+
+  // this provides basic logging (to stdout) until .start() is called below
+  // FIXME!!!
+  private val untypedSystem = a.ActorSystem(name + "-untyped", _config)
+  override def eventStream = untypedSystem.eventStream
+
+  override val logFilter: e.LoggingFilter = {
+    val arguments = Vector(classOf[Settings] → settings, classOf[e.EventStream] → eventStream)
+    dynamicAccess.createInstanceFor[e.LoggingFilter](settings.LoggingFilter, arguments).get
+  }
+
+  override val log: e.LoggingAdapter = new e.BusLogging(eventStream, getClass.getName + "(" + name + ")", this.getClass, logFilter)
+
+  /**
+   * Create the scheduler service. This one needs one special behavior: if
+   * Closeable, it MUST execute all outstanding tasks upon .close() in order
+   * to properly shutdown all dispatchers.
+   *
+   * Furthermore, this timer service MUST throw IllegalStateException if it
+   * cannot schedule a task. Once scheduled, the task MUST be executed. If
+   * executed upon close(), the task may execute before its timeout.
+   */
+  protected def createScheduler(): a.Scheduler =
+    dynamicAccess.createInstanceFor[a.Scheduler](settings.SchedulerClass, immutable.Seq(
+      classOf[Config] → settings.config,
+      classOf[e.LoggingAdapter] → log,
+      classOf[ThreadFactory] → threadFactory.withName(threadFactory.name + "-scheduler"))).get
+
+  override val scheduler: a.Scheduler = createScheduler()
+  private def closeScheduler(): Unit = scheduler match {
+    case x: Closeable ⇒ x.close()
+    case _            ⇒
+  }
+
+  override val dispatchers: Dispatchers = new DispatchersImpl(settings, log)
+  override val executionContext: ExecutionContextExecutor = dispatchers.lookup(DispatcherDefault)
+
+  override val startTime: Long = System.currentTimeMillis()
+  override def uptime: Long = (System.currentTimeMillis() - startTime) / 1000
+
+  private val terminationPromise: Promise[Terminated] = Promise()
+
+  private val rootPath: a.ActorPath = a.RootActorPath(a.Address("typed", name))
+
+  private val topLevelActors = new ConcurrentSkipListSet[ActorRefImpl[Nothing]]
```

Comment:
well, it’s two right now (`/user` and `/system`), but we can easily add more if needed